### DISCUSSION TITLE
Access collections based on variable

### DISCUSSION THREAD

**Discussion by danielLee0721** [11ty/eleventy#1771] [Q&A] [answered]
Hi there,

Is it possible to pass in variables to access collections?
ie. collections['someVariable']

I would like to be able to dynamically loop through collections based on a front matter variable.

Any help would be greatly appreciated!

Thank you,
D

**Comment by pdehaan** [comment, marked as answer]
`collections['someVariable']` should work, if I remember correctly.
Where are you trying to use a dynamic collection name that the syntax wasn't working?

**Reply by danielLee0721** [reply]
Thank you for the reply!

What I am trying to do is make an index page showing a list of links to all pages with a certain tag
So if I have a number of pages with the tag
tags: ['face'];


and then on my page I do
```
---
layout: index-list-page.html
type: face
---

```

and on the 'index-list-page' layout page I do 
```
---
layout: default.html
---
<div class="container">
  <ul class="index-list">
    {%- assign indexList = collections['type'] -%}
    {%- for i in indexList -%}
    <li>{{ i.url }}</li>
    {%- endfor -%}
  </ul>
</div>
{{ content }}

```

I was hoping that the layout page would be able to receive the 'type' variable from the template page.
Is this possible?

**Reply by pdehaan** [reply]
I was just poking at this, and here is my minimal sample:

```liquid
---
title: HOME
permalink: /
collectionName: posts
---

<h1>Posts</h1>

{% for p in collections[collectionName] %}
<p>{{ p.data.title }}</p>
{% endfor %}
```


Ah, but I see you just posted a comment... Yeah, you're really close! instead of `collections['type']`, it should be `collections[type]` (without the nested quotes). If you use quotes around `'type'`, it will look for a collection named "type" (`collections.type`) instead of `collections.face`.

**Reply by pdehaan** [reply]
Sorry, after seeing your response and sample code, I guess my initial reply from 18 hours ago was pretty confusing.

**Reply by danielLee0721** [reply]
Omg it worked! Thank you so much!! I thought I tried doing it without the '' but I guess not.
